ext/xml/xml.c

```diff
@@ -1091,15 +1091,19 @@ PHP_FUNCTION(xml_set_object)
 PHP_FUNCTION(xml_set_element_handler)
 {
 	xml_parser *parser;
-	zval *pind, *shdl, *ehdl;
+	zval *pind;
+	zend_fcall_info shdl_fci, ehdl_fci;
+	zend_fcall_info_cache shdl_fcc, ehdl_fcc;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "Ozz", &pind, xml_parser_ce, &shdl, &ehdl) == FAILURE) {
-		RETURN_THROWS();
-	}
+	ZEND_PARSE_PARAMETERS_START(3, 3)
+		Z_PARAM_OBJECT_OF_CLASS(pind, xml_parser_ce)
+		Z_PARAM_FUNC(shdl_fci, shdl_fcc)
+		Z_PARAM_FUNC(ehdl_fci, ehdl_fcc)
+	ZEND_PARSE_PARAMETERS_END();
 
 	parser = Z_XMLPARSER_P(pind);
-	xml_set_handler(&parser->startElementHandler, shdl);
-	xml_set_handler(&parser->endElementHandler, ehdl);
+	xml_set_handler(&parser->startElementHandler, &shdl_fci.function_name);
```

Comment:
I think it's even worse than that, you can also pass complex callables, but if you just pass a string it will be taken as an object method, not a function.

Reply:
:/ So no good solution here, right? Then I'll get rid of the related changes :(

Reply:
Is any longer-term action possible here? Like adding a deprecation notice in PHP 8.1 if the "callable" argument refers to a method, but comes only as a string?

Reply:
Sorry, I had to use `commit --amend` to get rid of the bogus callable changes. But the only changes I had were addressing the two comments.